### PR TITLE
Add InterfaceStatistics block and more.

### PR DESCRIPTION
**Pull request by ekse**
This pull request introduces the following changes:
- Add support for Interface Statistics blocks.
- Add enums for a couple constants.
- Add a couple tests.
- Put tests in a block and remove unused imports.
- Reformatted code with rustfmt.

I can split it you would prefer that.

### CODE REVIEW

File: examples/read_pcapng.rs [hunk 1]
```diff
@@ -19,10 +19,12 @@ fn main() {
     let mut buf: Vec<u8> = Vec::new();
     let _ = fh.read_to_end(&mut buf);
 
-    match pcapng::block::parse_blocks(&buf[..]) {
+    match parse_blocks(buf.as_slice()) {
```
Comment (richo) [inline examples/read_pcapng.rs:22]: I thought the &[..] syntax was preferred once that coersion landed? Am I missing something?

**Comment by richo** [comment]
This looks awesome. There are a couple things I want to digest a bit (mostly some of the interfaces changing to expose IResults, which I think is correct but want to ponder over. Should either merge or have feedback my monday. Thanks again!

File: examples/read_pcapng.rs (hunk 1 above)
Comment (ekse) [reply examples/read_pcapng.rs:22]: That's entirely possible, I will change it back to the way it was.

**Comment by ekse** [comment]
Friendly ping.

**Comment by richo-stripe** [comment]
Sorry for the delay. I'm gunna merge, if I change my mind about the IResult thing we can always go back.

Thanks again, and apologies!

**Pull request merged**